documentation/compute_environments.md

```diff
@@ -1,62 +1,70 @@
+There are two ways to view the software pre-installed in the containers:
 
-1. Open a terminal in the Fornax Science Console JupyterLab.
-2. Activate the environment you want to inspect using:
-   ```conda activate <environment-name>```
-3. Run the following command to list all installed packages:
-    ```conda list```
+- **Inside the Console**:
+The list of environment files (`.yml` for conda and requirements.txt for pip) of all the installed environments can be found in the folder `$LOCK_DIR` (defaults to /opt/lock). Each environment in the container has a corresponding file there. For instance:
+    - *requirements-python3.txt*: is the pip requirement file for the `python3` environment.
 
+    - *base-lock.yml*: is the conda environment file for the base conda environment running jupyter-lab
 
-For more details, see the official [conda list documentation](https://docs.conda.io/projects/conda/en/latest/commands/list.html).
 
-## Installing additional software on the Fornax Science Console
-
-There are two basic ways to install software that you would like to use in the Fornax Science Console, if the offereed environments do not meet your needs.
-
-### Create your own Conda Environment
-
-Creating your own Conda environment gives you full control to install custom packages, libraries, and tools—and your environment will be saved across sessions, even if you log out or restart your server.
-
-How to get started:
-
-1. Login to the Fornax Science Console, start JupyterHub, and open a terminal from the JupyterLab Launcher.
-
-2. Create a new Conda environment:
-
-```conda create --name myenv python=3.10```
+- **Outside the Console**: The same environment files are also available for every release of the container images on github. They are grouped by image name and available in the [container images release page](https://github.com/nasa-fornax/fornax-images/releases).
 
-3. Activate it:
+## Environment Selection
+`python3` is the default python environment. It has general astronomy and plotting software.
 
-```conda activate myenv```
+Each of the Fornax demo notebooks has its own environment with name of the `py-{notebook-name}` (e.g. `py-light_curve_generator` and `py-multiband_photometry`). When opening the notebook, the corresponding kernel should start automatically. You can also select it from the drop down kernel menu at the top-right of an open notebook.
 
-4. Install any additional packages you need:
+To activate a specific environment from the terminal, run: `source $ENV_DIR/{environment-name}/bin/activate`.
 
-```conda install numpy matplotlib astropy```
+For example, to activate the `py-light_curve_classifier` environment, run: 
+```sh
+source $ENV_DIR/py-light_curve_classifier/bin/activate
+```
+and the following to deactivate it.
+```sh
+deactivate
+```
 
-For more detailed help, check out the Conda documentation:
+There are two types of environments, **uv**-based and **conda**-based:
+- **uv**-based:
```

Comment:
I still wonder about pixi, arguably it doesn't (yet) support requirements files, but the workaround here should work: https://github.com/prefix-dev/pixi/issues/1410#issuecomment-2468549172

Reply:
yes, this partly works. It does not support nested requirements used in the notebook repo